pdns/pdns_recursor.cc

```diff
@@ -2080,10 +2081,10 @@ static void houseKeeping(void *)
     if(now.tv_sec - last_prune > (time_t)(5 + t_id)) {
       DTime dt;
       dt.setTimeval(now);
-      t_RC->doPrune(); // this function is local to a thread, so fine anyhow
-      t_packetCache->doPruneTo(::arg().asNum("max-packetcache-entries") / g_numWorkerThreads);
+      t_RC->doPrune(g_maxCacheEntries / g_numThreads); // this function is local to a thread, so fine anyhow
+      t_packetCache->doPruneTo(g_maxPacketCacheEntries / g_numWorkerThreads);
 
-      SyncRes::pruneNegCache(::arg().asNum("max-cache-entries") / (g_numWorkerThreads * 10));
+      SyncRes::pruneNegCache(g_maxCacheEntries / (g_numWorkerThreads * 10));
```

Comment:
Not related to this PR but I wonder where does the 10 comes from in `g_numWorkerThreads * 10`. And why we divide by `g_numWorkerThreads` (the number of _worker_ threads) for the packet and neg caches, but by `g_numThreads` (number of worker threads + 1 if `pdns-distribute-queries` is set) for the query cache.